app/models/signed_up_team.rb

```diff
@@ -81,4 +81,20 @@ def self.topic_id_by_team_id(team_id)
       signed_up_teams.first.topic_id
     end
   end
+
+  # Remove a specific signed_up_teams record for a given topic and team.
+  def self.drop_off_signup_record(topic_id,team_id)
+    # Fetching record for a given topic and team.
+    signup_record = SignedUpTeam.find_by(topic_id: topic_id, team_id: team_id)
+    # If the signup_record in not nil destroy it.
+    signup_record.destroy unless signup_record.nil?
+  end
+
+  # Remove all waitlisted records in signed_up_teams associated with a specific team.
+  def self.drop_off_waitlists(team_id)
```

Comment:
Why can't the team be the receiver here, and then the method cold be an instance method?